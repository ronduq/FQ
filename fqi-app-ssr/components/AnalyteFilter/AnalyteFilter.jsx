import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash.get';
import classNames from 'classnames';
import AnalyteReading from '../AnalyteReading/AnalyteReadingContainer';
import Tooltip from '../../hocs/TooltipContainer';

import s from '../TableFilter/TableFilter.scss';

class AnalyteFilter extends React.Component {

  constructor(props){
    super(props);
    this.state = {
      highlight: 'water',
    }
    this.handleClick = this.handleClick.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount() {
    const { produceInfo } = this.props;
    this.setState({
      highlight: get(produceInfo,['info','analytes'],[{}])[0].id,
    });
  }
  
  handleClick (analyte) {
    this.setState({
      highlight: analyte.id,
    });
  }

  handleChange (event) {
    const prod = event.target.value;
    const { produceInfo } = this.props;
    this.handleClick(produceInfo.info.analytes[prod]);
  }

  render (){
    const {
      produceInfo, 
      viewport,
      producesData,
      content,
    } = this.props;
    let selectedProduceInfo = {}

    Object.keys(get(producesData, 'itemsInfo')).map(prod => {
      if (producesData.itemsInfo[prod].id === get(produceInfo.info,'produceCode')) {
        selectedProduceInfo = producesData.itemsInfo[prod]
      }
    })

      return (
        <div className={s.container}>
          <div className={s.title}>
            <Tooltip 
              side='right' 
              size='large'
              bubblePosition='right'
              id='analyte_filter_title'
            >
              <h3>{content.analyte_breakdown}</h3>
            </Tooltip>
          </div>  
          <div className={s.body}>
            {viewport !== 'mobile' &&
            <ul className={s.filter} >
              { Object.entries(produceInfo.info).length > 0 &&
                Object.keys(get(produceInfo, ['info','analytes'])).map((prod, index) => {
                const analyte = produceInfo.info.analytes[prod];
                return (
                  <li 
                    id={`AnalyteFilterListItem${index}`} 
                    key={`AnalyteFilterListItem-${index}`} 
                    className={classNames({[s.highlight] : analyte.id === this.state.highlight })}
                  >
                    <button 
                      className={s.filterButton} 
                      onClick={() => this.handleClick(analyte)}
                    >
                      {analyte.label} <span className={s.buttonInfo}>{analyte.score}%</span>
                    </button>
                  </li>
                )
              })}
            </ul> }
            { viewport === 'mobile' &&
              Object.entries(produceInfo.info).length > 0 &&
              <div className={s.dropdownContainer}>
                <label className={s.dropdownLabel} htmlFor="produceList">{content.analytes}</label>
                <select className={s.dropdown} id="produceList" name="select" onChange={this.handleChange}>
                { Object.keys(get(produceInfo, 'info.analytes')).map((prod, index) => {
                    const analyte = produceInfo.info.analytes[prod];
                      return (
                      <option key={`AnalyteFilterListItem-${index}`} value={prod}>
                         {analyte.label}
                      </option>
                      )
                  })}
                </select> 
              </div>
            }  

            <div className={s.content}>
              <AnalyteReading 
                analyteId={this.state.highlight}
                produce={selectedProduceInfo}
              />
            </div>
          </div>
        </div>
      )}
};

AnalyteFilter.propTypes = {
  produceInfo: PropTypes.shape(), 
  producesData: PropTypes.shape(), 
  viewport: PropTypes.string,
};

AnalyteFilter.defaultProps = {
  produceInfo: {},
  producesData: {}, 
  viewport: '',
};

export default AnalyteFilter;
