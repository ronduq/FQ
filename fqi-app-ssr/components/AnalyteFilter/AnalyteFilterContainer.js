import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import AnalyteFilter from './AnalyteFilter';
import { 
  getProduceInfo,
  getProducesState,
  getViewport ,
  getUsdaContent,
  getProduceProfileContent,
} from '../../selectors/index';
import { setViewport } from '../../actions/content';

const mapStateToProps = state => {
  return {
    viewport: getViewport(state),
    produceInfo: getProduceInfo(state),
    producesData: getProducesState(state),
    usdaTexts: getUsdaContent(state),
    content: getProduceProfileContent(state),
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setViewport: (width) => setViewport(width),
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AnalyteFilter);