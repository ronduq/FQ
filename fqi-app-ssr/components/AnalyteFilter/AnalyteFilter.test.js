/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { mount } from 'enzyme'

import AnalyteFilter from './AnalyteFilter'

const fakeStore = {
  dispatch: jest.fn(),
  getState: () => ({
    produceInfo: {
      info: {
        analytes: [ 
        {score: 99.75, recentScan: 113, usda: 73, status: "Green", id: "sucrose"},
        {score: 98, recentScan: 123, usda: 70, status: "Green", id: "fructose"},
        {score: 92.5, recentScan: 119, usda: 77, status: "Green", id: "glucose"},
        {score: 65.25, recentScan: 46, usda: 69, status: "Red", id: "vitaminc"},
        {score: 94, recentScan: 117, usda: 70, status: "Green", id: "water"},
        ]
      },
      labels: [
        {label: "Water", id: "water"},
        {label: "Glucose", id: "glucose"},
        {label: "Sucrose", id: "sucrose"},
        {label: "Fructose", id: "fructose"},
        {label: "Citric Acid", id: "citric"},
      ]
    }, 
    producesData: {
      allItemsSelected: true,
      items: [
        {produceCode: "apple", produceName: "Apple", isParent: true, isChild: false, colour: "#00405C"},
        {produceCode: "apple-braeburn", produceName: "Apple Braeburn", isParent: false, isChild: true, parentCode: "apple"},
        {produceCode: "apple-fuji", produceName: "Apple Fuji", isParent: false, isChild: true, parentCode: "apple"},
        {produceCode: "apple-gala", produceName: "Apple Gala", isParent: false, isChild: true, parentCode: "apple"},
      ],
      itemsInfo: [
        {id: "apple", water: "", sucrose: "", vitaminc: "", label: "All Apples"},
        {id: "apple-braeburn", water: "Water in breaburn apples  text", sucrose: "sucrose in breaburn apples  text", vitaminc: "vitaminc in breaburn apples  text", label: "Braeburn Apples"},
        {id: "apple-fuji", water: "", sucrose: "", vitaminc: "", label: "Fuji Apples"},
        {id: "apple-gala", water: "", sucrose: "", vitaminc: "", label: "Gala Apples"},
        {id: "apple-honey-crisp", water: "", sucrose: "", vitaminc: "", label: "Honey Crisp Apples"},
        {id: "grape-green", water: "", sucrose: "", vitaminc: "", label: "Green Grapes"},
        {id: "grape-red", water: "", sucrose: "", vitaminc: "", label: "Red Grapes"},
        {id: "banana", water: "", sucrose: "", vitaminc: "", label: "Bananas"},
      ]
    },
    content: {
      usdaTexts: {
        green_title: 'test',
        green: 'test',
      }
    },
    content: {
      tooltips:{
        trend:{}
      }
    }
    // viewport: PropTypes.string,
  }),
  subscribe: jest.fn()
};

describe('AnalyteFilter', () => {

  const props = {
    content: {
      analyte_breakdown: 'test'
    },
    produceInfo: {
      info: {
        analytes: [ 
        {score: 99.75, recentScan: 113, usda: 73, status: "Green", id: "sucrose"},
        {score: 98, recentScan: 123, usda: 70, status: "Green", id: "fructose"},
        {score: 92.5, recentScan: 119, usda: 77, status: "Green", id: "glucose"},
        {score: 65.25, recentScan: 46, usda: 69, status: "Red", id: "vitaminc"},
        {score: 94, recentScan: 117, usda: 70, status: "Green", id: "water"},
        ]
      },
      labels: [
        {label: "Water", id: "water"},
        {label: "Glucose", id: "glucose"},
        {label: "Sucrose", id: "sucrose"},
        {label: "Fructose", id: "fructose"},
        {label: "Citric Acid", id: "citric"},
      ]
    }, 
    producesData: {
      allItemsSelected: true,
      items: [
        {produceCode: "apple", produceName: "Apple", isParent: true, isChild: false, colour: "#00405C"},
        {produceCode: "apple-braeburn", produceName: "Apple Braeburn", isParent: false, isChild: true, parentCode: "apple"},
        {produceCode: "apple-fuji", produceName: "Apple Fuji", isParent: false, isChild: true, parentCode: "apple"},
        {produceCode: "apple-gala", produceName: "Apple Gala", isParent: false, isChild: true, parentCode: "apple"},
      ],
      itemsInfo: [
        {id: "apple", water: "", sucrose: "", vitaminc: "", label: "All Apples"},
        {id: "apple-braeburn", water: "Water in breaburn apples  text", sucrose: "sucrose in breaburn apples  text", vitaminc: "vitaminc in breaburn apples  text", label: "Braeburn Apples"},
        {id: "apple-fuji", water: "", sucrose: "", vitaminc: "", label: "Fuji Apples"},
        {id: "apple-gala", water: "", sucrose: "", vitaminc: "", label: "Gala Apples"},
        {id: "apple-honey-crisp", water: "", sucrose: "", vitaminc: "", label: "Honey Crisp Apples"},
        {id: "grape-green", water: "", sucrose: "", vitaminc: "", label: "Green Grapes"},
        {id: "grape-red", water: "", sucrose: "", vitaminc: "", label: "Red Grapes"},
        {id: "banana", water: "", sucrose: "", vitaminc: "", label: "Bananas"},
      ]
    },
    // viewport: PropTypes.string,
  };

  it('AnalyteFilter snapshot"', () => {
    const component = renderer.create(
    <Provider store={fakeStore}>
      <AnalyteFilter {...props} />
    </Provider>);
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

  it('AnalyteFilter snapshot', () => {
    const wrapper = mount(
      <Provider store={fakeStore}>
        <AnalyteFilter {...props} />
      </Provider>
    );

    setTimeout(() => {
      const button = wrapper.find('AnalyteFilterListItem0');
    button.find('button').simulate('click');
    expect(props.handleClick).toHaveBeenCalledOnce();
    }, 100);
    
  });

  

})