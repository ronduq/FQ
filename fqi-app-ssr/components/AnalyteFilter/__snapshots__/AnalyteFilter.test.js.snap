// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyteFilter AnalyteFilter snapshot" 1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    <div
      className="wrapper"
    >
      <span
        className="content"
      >
        <h3>
          test
        </h3>
      </span>
      
    </div>
  </div>
  <div
    className="body"
  >
    <ul
      className="filter"
    >
      <li
        className="highlight"
        id="AnalyteFilterListItem0"
      >
        <button
          className="filterButton"
          onClick={[Function]}
        >
           
          <span
            className="buttonInfo"
          >
            99.75
            %
          </span>
        </button>
      </li>
      <li
        className=""
        id="AnalyteFilterListItem1"
      >
        <button
          className="filterButton"
          onClick={[Function]}
        >
           
          <span
            className="buttonInfo"
          >
            98
            %
          </span>
        </button>
      </li>
      <li
        className=""
        id="AnalyteFilterListItem2"
      >
        <button
          className="filterButton"
          onClick={[Function]}
        >
           
          <span
            className="buttonInfo"
          >
            92.5
            %
          </span>
        </button>
      </li>
      <li
        className=""
        id="AnalyteFilterListItem3"
      >
        <button
          className="filterButton"
          onClick={[Function]}
        >
           
          <span
            className="buttonInfo"
          >
            65.25
            %
          </span>
        </button>
      </li>
      <li
        className=""
        id="AnalyteFilterListItem4"
      >
        <button
          className="filterButton"
          onClick={[Function]}
        >
           
          <span
            className="buttonInfo"
          >
            94
            %
          </span>
        </button>
      </li>
    </ul>
    <div
      className="content"
    >
      <div
        className="container"
      >
        <ul
          className="readingMetrics"
        >
          <li>
            <h3>
               
            </h3>
          </li>
          <li
            className="metric"
          >
            <span
              className="metricTitle"
            />
             
            <span
              className="metricText"
            >
              99.75
            </span>
            <span
              className="metricSuffix"
            />
          </li>
          <li
            className="metric"
          >
            <span
              className="metricTitle"
            />
             
            <span
              className="metricText"
            >
              113
            </span>
            <span
              className="metricSuffix"
            />
          </li>
          <li
            className="metric"
          >
            <span
              className="metricTitle"
            />
             
            <span
              className="metricText"
            >
              73
            </span>
            <span
              className="metricSuffix"
            />
          </li>
        </ul>
        <div
          className="status"
        >
          <div
            className="statusLabel"
          />
          <div
            className="statusText"
          >
             
             
             
          </div>
        </div>
        <div>
          <h3>
             & 
          </h3>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "undefined",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
